stack-lts-14.yaml

```diff
@@ -17,7 +17,7 @@
 #
 # resolver: ./custom-snapshot.yaml
 # resolver: https://example.com/snapshots/2018-01-01.yaml
-resolver: lts-15.10
+resolver: lts-14.27
```

Comment:
I moved this back to lts-14.27 for the following two reasons:

- The file name is `stack-lts-14.yaml`, so I thought having this be lts-14.27 makes more sense.
- It is not possible to build `password` against lts-14 without specifying an `extra-dep`, but I don't think there is a way to do that from the command line, so it makes sense to keep around this `stack-lts-14.yaml` file, mostly for CI (assuming we do want to test on lts-14).